packages/semaphore-compat/semaphore-compat.1.0.2/opam

```diff
@@ -0,0 +1,41 @@
+opam-version: "2.0"
+synopsis: "Compatibility Semaphore module"
+description: """
+Projects that want to use the Semaphore module defined in OCaml 4.12.0 while
+staying compatible with older versions of OCaml should use this library
+instead.
+"""
+maintainer: ["Craig Ferguson <me@craigfe.io>"]
+authors: ["Xavier Leroy"]
+license: "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
+homepage: "https://github.com/mirage/semaphore-compat"
+doc: "https://mirage.github.io/semaphore-compat"
+bug-reports: "https://github.com/mirage/semaphore-compat/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml"
+]
+build: [
+  ["dune" "subst"] {pinned}
```

Comment:
To address the lint warning

```
Warning in semaphore-compat.1.0.2: Dubious use of 'dune subst'. 'dune subst' should always only be called with {dev} (i.e. ["dune" "subst"] {dev}) If your opam file has been autogenerated by dune, you need to upgrade your dune-project to at least (lang dune 2.7).
```

Perhaps this should be

```suggestion
  ["dune" "subst"] {dev}
```

as per https://github.com/ocaml/dune/pull/3647 and https://github.com/ocaml/opam-repository/wiki/PR-checks#some-common-failures

Reply:
hum this file is generated by dune  ...

Reply:
Based on the lint message, I guess the issue is that the `dune-project` file is still specifying an old dune version: https://github.com/mirage/semaphore-compat/blob/ee495458d0f2ed5c6f4e0ba3dbcf8c37d5b767d4/dune-project#L1